src/Runtime/OMTensor.inc

```diff
@@ -526,14 +527,33 @@ OMTensor *omTensorCreateWithShape(const std::vector<int64_t> &shape) {
   return omt;
 }
 
+// Static variables used by omDefineSeed and omTensorCreateWithRandomData.
+static unsigned int omUseOneSeed = 0;
+static std::mt19937 omRandomGenerator(0);
+
+// When called, a single seed will be used, either from the given seed or from
+// another random number generator.
+unsigned int omDefineSeed(unsigned int seed, unsigned int ignoreSeed) {
```

Comment:
[Suggestion]: use an Optional to indicate that the `seed` might not be provided ? As in: 
```
unsigned omDefineSeed(Optional<unsigned> optionalSeed) {
  std::random_device rd;
  unsigned seed = optionalSeed.hasValue() ? optionalSeed.getValue() : rd();
  ...
```

Reply:
While I agree that the optional is much more elegant, currently the file has no dependence to LLVM, so adding `Optional` just for this use seems like a big change. I will change "ignoreSeed" to `hasSeedValue` so that it may feel more like the optional interface.